taskvine/src/bindings/python3/ndcctools/taskvine/futures.py

```diff
@@ -142,13 +142,52 @@ def _iterator():
     return _iterator()
 
 
+def run_iterable(fn, iterable, dimensions=1):
+
+    if not ((hasattr(iterable, '__iter__') or hasattr(iterable, '__getitem__')) and not isinstance(iterable, str)):
```

Comment:
`from collections.abc import Sequence` and then `isinstance(iterable, Sequence)`. I would remove the special case for the string, as you are not checking for other similar objects (byte arrays, etc.)

I'm not sure why you need `__getitem__`. If it is not needed, you can use `Iterable` instead of `Sequence` above.